File: Module1 - Zachary Abbe -CS499/Project1-ZacharyAbbe-CS320/Task.java
/*****************************
 * Name: Zachary Abbe
 * Class: CS320
 * Date: 04/12/2025
 * Desc: This contains the Task object containing the id, name and desc
 ****************************/



public class Task {

    private String uniqId;
    private String fullName;
    private String desc;

    // constructor for the object
    public Task(String Id, String name, String desc){
        setID(Id);
        setFullName(name);
        setDesc(desc);

    }

    //setters for the object
    private void setID(String id){
        if (id == null || id.isEmpty() || id.length() >10){
            throw new IllegalArgumentException("Invalid ID");
        }
        this.uniqId=id;
    }

    public void setFullName(String name){
        if (name==null || name.isEmpty() || name.length() > 20){

            throw new IllegalArgumentException("Invalid name");
        }
        this.fullName = name;
    }

    public void setDesc(String desc1) {
        if (desc1 == null || desc1.isEmpty() || desc1.length() > 50) {

            throw new IllegalArgumentException("Invalid desc");
        }
        this.desc = desc1;
    }
    //getters
    public String getUniqId(){
        return uniqId;
    }
    public String getFullName(){
        return fullName;
    }
    public String getDesc(){
        return desc;
    }

}

File: CS499 Project1 - ZacharyAbbe/Task.cpp
#include "Task.h"
#include <stdexcept>

using namespace std;

const int idLength = 10;
const int namelength = 20;
const int descripLength = 50;
Task::Task(const string& id, const string& name, const string& desc)
{
	setID(id);
	setFullName(name);
	setDesc(desc);
}

void Task::setID(const string& id)
{
	if (id.empty() || id.length() > idLength)
	{
		throw invalid_argument("Invalid ID");
	}
	this->uniqId = id;
}

void Task::setFullName(const string& name) 
{
	if (name.empty() || name.length() > namelength) 
	{
		throw invalid_argument("Invalid name");
	}
	this->fullName = name;
}

void Task::setDesc(const string& desc) 
{
	if (desc.empty() || desc.length() > descripLength) 
	{
		throw invalid_argument("Invalid desc");
	}
	this->desc = desc;
}


string Task::getUniqId() const 
{
	return uniqId;
}

string Task::getFullName() const 
{
	return fullName;
}

string Task::getDesc() const 
{
	return desc;
}